File: Apache_Simulator/Source/Apache_Simulator/Helicopter Logic/ApacheEngine.cpp
// Fill out your copyright notice in the Description page of Project Settings.


#include "ApacheEngine.h"

// Sets default values for this component's properties
UApacheEngine::UApacheEngine()
{
	// Set this component to be initialized when the game starts, and to be ticked every frame.  You can turn these features
	// off to improve performance if you don't need them.
	PrimaryComponentTick.bCanEverTick = true;


	// ...
}


// Called when the game starts
void UApacheEngine::BeginPlay()
{
	Super::BeginPlay();

	// ...
	
}


// Called every frame
void UApacheEngine::TickComponent(float DeltaTime, ELevelTick TickType, FActorComponentTickFunction* ThisTickFunction)
{
	Super::TickComponent(DeltaTime, TickType, ThisTickFunction);

	Power = FMath::Clamp(Power, 0.0f, 50.0f);

	if (IsOn)
		EngineOn();
	else
		EngineOff();
}

void UApacheEngine::EngineOn()
{
	
	Power += 0.025f;
}
void UApacheEngine::EngineOff()
{
	Power -= 0.08f;
}

File: ApacheSim/Assets/Scripts/HelicopterLogic/ApacheEngine.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class ApacheEngine : MonoBehaviour
{
  public  bool IsOn = false;
    public float Power = 0.0f;
    public float Damage;
    // Start is called before the first frame update
    void Start()
    {
        IsOn = false;
    }

    // Update is called once per frame
    void Update()
    {
        Power = Mathf.Clamp(Power, 0.0f, 50.0f);

        if (IsOn)
            EngineOn();
        else
            EngineOff();
    }
    void EngineOn()
    {

        Power += 0.025f;
    }

    void EngineOff()
    {
        Power -= 0.08f;
    }
}
